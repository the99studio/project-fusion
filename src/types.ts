/**
 * Type definitions for the fusion functionality
 */

// Branded types for better type safety
export type FilePath = string & { readonly __brand: unique symbol };
export type FileHash = string & { readonly __brand: unique symbol };
export type FileExtension = `.${string}`;

// Helper functions for branded types
export const createFilePath = (path: string): FilePath => path as FilePath;
export const createFileHash = (hash: string): FileHash => hash as FileHash;

/**
 * Main configuration interface
 */
export interface Config {
    fusion: {
        fusion_file: string;
        fusion_log: string;
        copyToClipboard: boolean;
    };
    parsedFileExtensions: {
        backend: string[];
        config: string[];
        cpp: string[];
        scripts: string[];
        web: string[];
        godot: string[];
        [key: string]: string[];
    };
    parsing: {
        parseSubDirectories: boolean;
        rootDirectory: string;
    };
    ignorePatterns: string[];
    useGitIgnoreForExcludes: boolean;
    schemaVersion: number;
}

/**
 * Information about a file for fusion
 */
export interface FileInfo {
    path: FilePath;
    content: string;
    hash: FileHash;
}

/**
 * Options for the fusion process
 */
export interface FusionOptions {
    extensionGroups?: string[];
}

/**
 * Result of the fusion process - Using discriminated union for better type safety
 */
export type FusionResult = 
    | {
        success: true;
        message: string;
        fusionFilePath: FilePath;
        logFilePath: FilePath;
    }
    | {
        success: false;
        message: string;
        logFilePath?: FilePath;
        error?: Error;
    };