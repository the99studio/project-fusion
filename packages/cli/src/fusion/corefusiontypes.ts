/**
 * Type definitions for the fusion functionality
 */

/**
 * Configuration for the fusion process
 */
export interface FusionConfig {
    directory: string;
    fusion_file: string;
    fusion_log: string;
  }
  
  /**
   * Configuration for parsed file extensions
   */
  export interface ParsedFileExtensionsConfig {
    backend: string[];
    config: string[];
    cpp: string[];
    scripts: string[];
    web: string[];
    [key: string]: string[];
  }
  
  /**
   * Configuration for parsing
   */
  export interface ParsingConfig {
    parseSubDirectories: boolean;
    rootDirectory: string;
  }
  
  /**
   * Information about a file for fusion
   */
  export interface FileInfo {
    path: string;
    content: string;
    hash: string;
  }
  
  /**
   * Options for the fusion process
   */
  export interface FusionOptions {
    extensionGroups?: string[];
  }
  
  /**
   * Result of the fusion process
   */
  export interface FusionResult {
    success: boolean;
    message: string;
    fusionFilePath?: string;
    logFilePath?: string;
    error?: Error;
  }