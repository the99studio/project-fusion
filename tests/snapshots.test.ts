// SPDX-License-Identifier: MIT
// Copyright (c) 2025 the99studio
/**
 * Snapshot tests for verifying MD/HTML format output
 */
import { existsSync } from 'node:fs';
import { writeFile, mkdir, rm, readFile } from 'node:fs/promises';
import { join } from 'node:path';
import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { processFusion } from '../src/fusion.js';
import { defaultConfig } from '../src/utils.js';
import { normalizeFilePaths } from './test-helpers.js';

describe('Format Snapshot Tests', () => {
    const testDir = join(process.cwd(), 'temp', 'snapshot-test');
    const originalCwd = process.cwd();

    beforeEach(async () => {
        // Clean up and create test directory
        if (existsSync(testDir)) {
            await rm(testDir, { recursive: true, force: true });
        }
        await mkdir(testDir, { recursive: true });
        process.chdir(testDir);
    });

    afterEach(async () => {
        process.chdir(originalCwd);
        if (existsSync(testDir)) {
            await rm(testDir, { recursive: true, force: true });
        }
    });

    describe('Markdown Format Snapshots', () => {
        it('should generate consistent markdown format for JavaScript files', async () => {
            // Create sample JavaScript files
            await writeFile('index.js', `// Main application entry point
const express = require('express');
const app = express();

app.get('/', (req, res) => {
    res.send('Hello, World!');
});

app.listen(3000, () => {
    console.log('Server running on port 3000');
});`);

            await writeFile('utils.js', `// Utility functions
function formatDate(date) {
    return date.toISOString().split('T')[0];
}

function capitalize(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
}

module.exports = { formatDate, capitalize };`);

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: true,
                generateHtml: false,
                generatedFileName: 'test-fusion',
                parsedFileExtensions: {
                    web: ['.js']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const mdContent = await readFile('test-fusion.md', 'utf8');
            
            // Normalize timestamps and paths for consistent snapshots
            const normalizedMd = mdContent
                .replaceAll(/\*\*Generated:\*\* [^\n]+/g, '**Generated:** [TIMESTAMP]')
            
            const pathNormalizedMd = normalizeFilePaths(normalizedMd);
            
            // Test structure and content
            expect(pathNormalizedMd).toMatchSnapshot('javascript-files.md');
        });

        it('should generate consistent markdown format for TypeScript files', async () => {
            await writeFile('types.ts', `// Type definitions
export interface User {
    id: number;
    name: string;
    email: string;
    createdAt: Date;
}

export type UserRole = 'admin' | 'user' | 'moderator';

export interface CreateUserRequest {
    name: string;
    email: string;
    role?: UserRole;
}`);

            await writeFile('service.ts', `// User service
import { User, CreateUserRequest, UserRole } from './types.js';

export class UserService {
    private users: User[] = [];
    private nextId = 1;

    createUser(request: CreateUserRequest): User {
        const user: User = {
            id: this.nextId++,
            name: request.name,
            email: request.email,
            createdAt: new Date()
        };
        
        this.users.push(user);
        return user;
    }

    findUser(id: number): User | undefined {
        return this.users.find(user => user.id === id);
    }
}`);

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: true,
                generateHtml: false,
                generatedFileName: 'typescript-fusion',
                parsedFileExtensions: {
                    web: ['.ts']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const mdContent = await readFile('typescript-fusion.md', 'utf8');
            
            // Normalize timestamps and paths for consistent snapshots
            const normalizedMd = mdContent
                .replaceAll(/\*\*Generated:\*\* [^\n]+/g, '**Generated:** [TIMESTAMP]');
                
            const pathNormalizedMd = normalizeFilePaths(normalizedMd);
            expect(pathNormalizedMd).toMatchSnapshot('typescript-files.md');
        });

        it('should generate consistent markdown format for mixed file types', async () => {
            await writeFile('config.json', `{
  "name": "test-project",
  "version": "1.0.0",
  "dependencies": {
    "express": "^4.18.0",
    "typescript": "^5.0.0"
  },
  "scripts": {
    "start": "node index.js",
    "build": "tsc"
  }
}`);

            await writeFile('README.md', `# Test Project

This is a test project for snapshot testing.

## Features

- **Fast**: Built with performance in mind
- **Secure**: Follows security best practices
- **Reliable**: Comprehensive test coverage

## Installation

\`\`\`bash
npm install
npm start
\`\`\`

## License

MIT License`);

            await writeFile('script.sh', `#!/bin/bash
# Deployment script

echo "Starting deployment..."

# Build the project
npm run build

# Run tests
npm test

# Deploy to production
echo "Deploying to production..."
rsync -av dist/ user@server:/var/www/app/

echo "Deployment complete!"`);

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: true,
                generateHtml: false,
                generatedFileName: 'mixed-fusion',
                parsedFileExtensions: {
                    config: ['.json'],
                    doc: ['.md'],
                    scripts: ['.sh']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const mdContent = await readFile('mixed-fusion.md', 'utf8');
            
            // Normalize timestamps and paths for consistent snapshots
            const normalizedMd = mdContent
                .replaceAll(/\*\*Generated:\*\* [^\n]+/g, '**Generated:** [TIMESTAMP]');
                
            const pathNormalizedMd = normalizeFilePaths(normalizedMd);
            expect(pathNormalizedMd).toMatchSnapshot('mixed-files.md');
        });
    });

    describe('HTML Format Snapshots', () => {
        it('should generate consistent HTML format for JavaScript files', async () => {
            await writeFile('app.js', `// Simple Express application
const express = require('express');
const path = require('path');

const app = express();
const PORT = process.env.PORT || 3000;

// Middleware
app.use(express.json());
app.use(express.static('public'));

// Routes
app.get('/', (req, res) => {
    res.sendFile(path.join(__dirname, 'index.html'));
});

app.get('/api/health', (req, res) => {
    res.json({ status: 'ok', timestamp: new Date().toISOString() });
});

// Start server
app.listen(PORT, () => {
    console.log(\`Server running on http://localhost:\${PORT}\`);
});`);

            await writeFile('helpers.js', `// Helper functions
const crypto = require('crypto');

/**
 * Generate a random ID
 * @param {number} length - Length of the ID
 * @returns {string} Random ID
 */
function generateId(length = 8) {
    return crypto.randomBytes(length).toString('hex');
}

/**
 * Validate email format
 * @param {string} email - Email to validate
 * @returns {boolean} True if valid
 */
function isValidEmail(email) {
    const regex = /^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$/;
    return regex.test(email);
}

module.exports = {
    generateId,
    isValidEmail
};`);

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: false,
                generateHtml: true,
                generatedFileName: 'html-test',
                parsedFileExtensions: {
                    web: ['.js']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const htmlContent = await readFile('html-test.html', 'utf8');
            
            // Normalize timestamps and paths for consistent snapshots
            const normalizedHtml = htmlContent
                .replaceAll(/<p><strong>Generated:<\/strong> [^<]+<\/p>/g, '<p><strong>Generated:</strong> TIMESTAMP</p>');
            
            const pathNormalizedHtml = normalizeFilePaths(normalizedHtml);
            expect(pathNormalizedHtml).toMatchSnapshot('javascript-files.html');
        });

        it('should generate consistent HTML format with proper escaping', async () => {
            await writeFile('template.html', `<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Test Template</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            margin: 0;
            padding: 20px;
        }
        .container {
            max-width: 800px;
            margin: 0 auto;
        }
        .header {
            background: #f5f5f5;
            padding: 20px;
            border-radius: 8px;
        }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1>Welcome to Test App</h1>
            <p>This is a <strong>test</strong> application with <em>HTML</em> content.</p>
        </div>
        
        <main>
            <h2>Features & Benefits</h2>
            <ul>
                <li>Fast & reliable</li>
                <li>Secure by design</li>
                <li>Easy to use</li>
            </ul>
            
            <p>Contact us at: <a href="mailto:test@example.com">test@example.com</a></p>
        </main>
    </div>
    
    <script>
        console.log('Page loaded successfully!');
        document.addEventListener('DOMContentLoaded', () => {
            console.log('DOM ready');
        });
    </script>
</body>
</html>`);

            await writeFile('styles.css', `/* Global styles */
* {
    box-sizing: border-box;
}

body {
    margin: 0;
    padding: 0;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', sans-serif;
    line-height: 1.6;
    color: #333;
}

.container {
    max-width: 1200px;
    margin: 0 auto;
    padding: 0 20px;
}

/* Button styles */
.btn {
    display: inline-block;
    padding: 12px 24px;
    background: #007bff;
    color: white;
    text-decoration: none;
    border-radius: 4px;
    transition: background 0.3s ease;
}

.btn:hover {
    background: #0056b3;
}

/* Responsive design */
@media (max-width: 768px) {
    .container {
        padding: 0 10px;
    }
    
    .btn {
        display: block;
        text-align: center;
        margin: 10px 0;
    }
}`);

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: false,
                generateHtml: true,
                generatedFileName: 'html-escape-test',
                parsedFileExtensions: {
                    web: ['.html', '.css']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const htmlContent = await readFile('html-escape-test.html', 'utf8');
            
            // Normalize timestamps and paths
            const normalizedHtml = htmlContent
                .replaceAll(/<p><strong>Generated:<\/strong> [^<]+<\/p>/g, '<p><strong>Generated:</strong> TIMESTAMP</p>');
            
            const pathNormalizedHtml = normalizeFilePaths(normalizedHtml);
            expect(pathNormalizedHtml).toMatchSnapshot('html-with-escaping.html');
        });

        it('should generate HTML with proper table of contents structure', async () => {
            await mkdir('api', { recursive: true });
            await mkdir('components', { recursive: true });
            await mkdir('utils', { recursive: true });
            
            await writeFile('api/users.js', 'const users = [];');
            await writeFile('api/posts.js', 'const posts = [];');
            await writeFile('components/Header.js', 'export default function Header() {}');
            await writeFile('components/Footer.js', 'export default function Footer() {}');
            await writeFile('utils/database.js', 'class Database {}');

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: false,
                generateHtml: true,
                generatedFileName: 'toc-test',
                parsedFileExtensions: {
                    web: ['.js']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const htmlContent = await readFile('toc-test.html', 'utf8');
            
            // Check TOC structure with simplified HTML format
            expect(htmlContent).toContain('<h2>Table of Contents</h2>');
            expect(htmlContent).toContain('<ul>');
            expect(htmlContent).toContain('href="#apiusersjs"');
            expect(htmlContent).toContain('href="#componentsheaderjs"');
            
            // Normalize timestamps and paths for consistent snapshots
            const normalizedHtml = htmlContent
                .replaceAll(/<p><strong>Generated:<\/strong> [^<]+<\/p>/g, '<p><strong>Generated:</strong> TIMESTAMP</p>');
            
            const pathNormalizedHtml = normalizeFilePaths(normalizedHtml);
            expect(pathNormalizedHtml).toMatchSnapshot('html-with-toc.html');
        });
    });

    describe('Cross-Format Consistency', () => {
        it('should maintain content consistency between markdown and HTML formats', async () => {
            await writeFile('example.ts', `// Example TypeScript module
export interface Config {
    apiUrl: string;
    timeout: number;
    retries: number;
}

export class ApiClient {
    constructor(private config: Config) {}
    
    async get<T>(path: string): Promise<T> {
        const response = await fetch(\`\${this.config.apiUrl}\${path}\`);
        if (!response.ok) {
            throw new Error(\`HTTP \${response.status}\`);
        }
        return response.json();
    }
}`);

            const config = {
                ...defaultConfig,
                rootDirectory: testDir,
                generateText: false,
                generateMarkdown: true,
                generateHtml: true,
                generatedFileName: 'consistency-test',
                parsedFileExtensions: {
                    web: ['.ts']
                }
            };

            const result = await processFusion(config);
            expect(result.success).toBe(true);

            const mdContent = await readFile('consistency-test.md', 'utf8');
            const htmlContent = await readFile('consistency-test.html', 'utf8');

            // Both should contain the same source code
            expect(mdContent).toContain('export interface Config');
            expect(htmlContent).toContain('export interface Config');
            
            expect(mdContent).toContain('export class ApiClient');
            expect(htmlContent).toContain('export class ApiClient');
            
            // Both should reference the same file
            expect(mdContent).toContain('example.ts');
            expect(htmlContent).toContain('example.ts');
            
            // Both should have proper structure
            expect(mdContent).toContain('## example.ts');
            expect(htmlContent).toContain('<h2 id="examplets">example.ts</h2>');
        });
    });
});