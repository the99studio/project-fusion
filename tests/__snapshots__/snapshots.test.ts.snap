// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Format Snapshot Tests > HTML Format Snapshots > should generate HTML with proper table of contents structure > html-with-toc.html 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="Content-Security-Policy" content="default-src 'none'; style-src 'unsafe-inline'; font-src 'self'; base-uri 'none'">
<meta http-equiv="X-Content-Type-Options" content="nosniff">
<meta http-equiv="X-Frame-Options" content="DENY">
<meta http-equiv="Referrer-Policy" content="no-referrer">
<title>snapshot-test - Project Fusion</title>
<style>
body{font-family:monospace;margin:20px;line-height:1.6;color:#000;background:#fff}
pre{background:#f5f5f5;border:1px solid #ccc;padding:10px;overflow-x:auto;white-space:pre}
h1,h2{margin-top:20px}
ul{padding-left:20px}
a{color:#00e;text-decoration:underline}
.error{background:#fee;border:1px solid #c00;padding:10px}
</style>
</head>
<body>
<h1>Project Fusion Output</h1>
<p><strong>Project:</strong> snapshot-test</p>
<p><strong>Generated:</strong> TIMESTAMP</p>
<p><strong>Files:</strong> 5</p>
<p><strong>Created with:</strong> <a href="https://github.com/the99studio/project-fusion" target="_blank" rel="noopener noreferrer">project-fusion v1.0.0</a></p>
<hr>
<h2>Table of Contents</h2>
<ul>
<li><a href="#apipostsjs">api/posts.js</a></li>
<li><a href="#apiusersjs">api/users.js</a></li>
<li><a href="#componentsfooterjs">components/Footer.js</a></li>
<li><a href="#componentsheaderjs">components/Header.js</a></li>
<li><a href="#utilsdatabasejs">utils/database.js</a></li>
</ul>
<hr>
<h2 id="apipostsjs">api/posts.js</h2>
<pre>const posts &#61; &#91;&#93;;</pre>
<h2 id="apiusersjs">api/users.js</h2>
<pre>const users &#61; &#91;&#93;;</pre>
<h2 id="componentsfooterjs">components/Footer.js</h2>
<pre>export default function Footer&#40;&#41; &#123;&#125;</pre>
<h2 id="componentsheaderjs">components/Header.js</h2>
<pre>export default function Header&#40;&#41; &#123;&#125;</pre>
<h2 id="utilsdatabasejs">utils/database.js</h2>
<pre>class Database &#123;&#125;</pre>
</body>
</html>"
`;

exports[`Format Snapshot Tests > HTML Format Snapshots > should generate consistent HTML format for JavaScript files > javascript-files.html 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="Content-Security-Policy" content="default-src 'none'; style-src 'unsafe-inline'; font-src 'self'; base-uri 'none'">
<meta http-equiv="X-Content-Type-Options" content="nosniff">
<meta http-equiv="X-Frame-Options" content="DENY">
<meta http-equiv="Referrer-Policy" content="no-referrer">
<title>snapshot-test - Project Fusion</title>
<style>
body{font-family:monospace;margin:20px;line-height:1.6;color:#000;background:#fff}
pre{background:#f5f5f5;border:1px solid #ccc;padding:10px;overflow-x:auto;white-space:pre}
h1,h2{margin-top:20px}
ul{padding-left:20px}
a{color:#00e;text-decoration:underline}
.error{background:#fee;border:1px solid #c00;padding:10px}
</style>
</head>
<body>
<h1>Project Fusion Output</h1>
<p><strong>Project:</strong> snapshot-test</p>
<p><strong>Generated:</strong> TIMESTAMP</p>
<p><strong>Files:</strong> 2</p>
<p><strong>Created with:</strong> <a href="https://github.com/the99studio/project-fusion" target="_blank" rel="noopener noreferrer">project-fusion v1.0.0</a></p>
<hr>
<h2>Table of Contents</h2>
<ul>
<li><a href="#appjs">app.js</a></li>
<li><a href="#helpersjs">helpers.js</a></li>
</ul>
<hr>
<h2 id="appjs">app.js</h2>
<pre>&#47;&#47; Simple Express application
const express &#61; require&#40;&#39;express&#39;&#41;;
const path &#61; require&#40;&#39;path&#39;&#41;;

const app &#61; express&#40;&#41;;
const PORT &#61; process.env.PORT || 3000;

&#47;&#47; Middleware
app.use&#40;express.json&#40;&#41;&#41;;
app.use&#40;express.static&#40;&#39;public&#39;&#41;&#41;;

&#47;&#47; Routes
app.get&#40;&#39;&#47;&#39;, &#40;req, res&#41; &#61;&gt; &#123;
    res.sendFile&#40;path.join&#40;__dirname, &#39;index.html&#39;&#41;&#41;;
&#125;&#41;;

app.get&#40;&#39;&#47;api&#47;health&#39;, &#40;req, res&#41; &#61;&gt; &#123;
    res.json&#40;&#123; status: &#39;ok&#39;, timestamp: new Date&#40;&#41;.toISOString&#40;&#41; &#125;&#41;;
&#125;&#41;;

&#47;&#47; Start server
app.listen&#40;PORT, &#40;&#41; &#61;&gt; &#123;
    console.log&#40;&#96;Server running on http:&#47;&#47;localhost:&#36;&#123;PORT&#125;&#96;&#41;;
&#125;&#41;;</pre>
<h2 id="helpersjs">helpers.js</h2>
<pre>&#47;&#47; Helper functions
const crypto &#61; require&#40;&#39;crypto&#39;&#41;;

&#47;**
 * Generate a random ID
 * &#64;param &#123;number&#125; length - Length of the ID
 * &#64;returns &#123;string&#125; Random ID
 *&#47;
function generateId&#40;length &#61; 8&#41; &#123;
    return crypto.randomBytes&#40;length&#41;.toString&#40;&#39;hex&#39;&#41;;
&#125;

&#47;**
 * Validate email format
 * &#64;param &#123;string&#125; email - Email to validate
 * &#64;returns &#123;boolean&#125; True if valid
 *&#47;
function isValidEmail&#40;email&#41; &#123;
    const regex &#61; &#47;^&#91;^/s&#64;&#93;&#43;&#64;&#91;^/s&#64;&#93;&#43;/.&#91;^/s&#64;&#93;&#43;&#36;&#47;;
    return regex.test&#40;email&#41;;
&#125;

module.exports &#61; &#123;
    generateId,
    isValidEmail
&#125;;</pre>
</body>
</html>"
`;

exports[`Format Snapshot Tests > HTML Format Snapshots > should generate consistent HTML format with proper escaping > html-with-escaping.html 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="Content-Security-Policy" content="default-src 'none'; style-src 'unsafe-inline'; font-src 'self'; base-uri 'none'">
<meta http-equiv="X-Content-Type-Options" content="nosniff">
<meta http-equiv="X-Frame-Options" content="DENY">
<meta http-equiv="Referrer-Policy" content="no-referrer">
<title>snapshot-test - Project Fusion</title>
<style>
body{font-family:monospace;margin:20px;line-height:1.6;color:#000;background:#fff}
pre{background:#f5f5f5;border:1px solid #ccc;padding:10px;overflow-x:auto;white-space:pre}
h1,h2{margin-top:20px}
ul{padding-left:20px}
a{color:#00e;text-decoration:underline}
.error{background:#fee;border:1px solid #c00;padding:10px}
</style>
</head>
<body>
<h1>Project Fusion Output</h1>
<p><strong>Project:</strong> snapshot-test</p>
<p><strong>Generated:</strong> TIMESTAMP</p>
<p><strong>Files:</strong> 2</p>
<p><strong>Created with:</strong> <a href="https://github.com/the99studio/project-fusion" target="_blank" rel="noopener noreferrer">project-fusion v1.0.0</a></p>
<hr>
<h2>Table of Contents</h2>
<ul>
<li><a href="#stylescss">styles.css</a></li>
<li><a href="#templatehtml">template.html</a></li>
</ul>
<hr>
<h2 id="stylescss">styles.css</h2>
<pre>&#47;* Global styles *&#47;
* &#123;
    box-sizing: border-box;
&#125;

body &#123;
    margin: 0;
    padding: 0;
    font-family: -apple-system, BlinkMacSystemFont, &#39;Segoe UI&#39;, sans-serif;
    line-height: 1.6;
    color: #333;
&#125;

.container &#123;
    max-width: 1200px;
    margin: 0 auto;
    padding: 0 20px;
&#125;

&#47;* Button styles *&#47;
.btn &#123;
    display: inline-block;
    padding: 12px 24px;
    background: #007bff;
    color: white;
    text-decoration: none;
    border-radius: 4px;
    transition: background 0.3s ease;
&#125;

.btn:hover &#123;
    background: #0056b3;
&#125;

&#47;* Responsive design *&#47;
&#64;media &#40;max-width: 768px&#41; &#123;
    .container &#123;
        padding: 0 10px;
    &#125;
    
    .btn &#123;
        display: block;
        text-align: center;
        margin: 10px 0;
    &#125;
&#125;</pre>
<h2 id="templatehtml">template.html</h2>
<pre>&lt;&#33;DOCTYPE html&gt;
&lt;html lang&#61;&quot;en&quot;&gt;
&lt;head&gt;
    &lt;meta charset&#61;&quot;UTF-8&quot;&gt;
    &lt;meta name&#61;&quot;viewport&quot; content&#61;&quot;width&#61;device-width, initial-scale&#61;1.0&quot;&gt;
    &lt;title&gt;Test Template&lt;&#47;title&gt;
    &lt;style&gt;
        body &#123;
            font-family: Arial, sans-serif;
            margin: 0;
            padding: 20px;
        &#125;
        .container &#123;
            max-width: 800px;
            margin: 0 auto;
        &#125;
        .header &#123;
            background: #f5f5f5;
            padding: 20px;
            border-radius: 8px;
        &#125;
    &lt;&#47;style&gt;
&lt;&#47;head&gt;
&lt;body&gt;
    &lt;div class&#61;&quot;container&quot;&gt;
        &lt;div class&#61;&quot;header&quot;&gt;
            &lt;h1&gt;Welcome to Test App&lt;&#47;h1&gt;
            &lt;p&gt;This is a &lt;strong&gt;test&lt;&#47;strong&gt; application with &lt;em&gt;HTML&lt;&#47;em&gt; content.&lt;&#47;p&gt;
        &lt;&#47;div&gt;
        
        &lt;main&gt;
            &lt;h2&gt;Features &amp; Benefits&lt;&#47;h2&gt;
            &lt;ul&gt;
                &lt;li&gt;Fast &amp; reliable&lt;&#47;li&gt;
                &lt;li&gt;Secure by design&lt;&#47;li&gt;
                &lt;li&gt;Easy to use&lt;&#47;li&gt;
            &lt;&#47;ul&gt;
            
            &lt;p&gt;Contact us at: &lt;a href&#61;&quot;mailto:test&#64;example.com&quot;&gt;test&#64;example.com&lt;&#47;a&gt;&lt;&#47;p&gt;
        &lt;&#47;main&gt;
    &lt;&#47;div&gt;
    
    &lt;script&gt;
        console.log&#40;&#39;Page loaded successfully&#33;&#39;&#41;;
        document.addEventListener&#40;&#39;DOMContentLoaded&#39;, &#40;&#41; &#61;&gt; &#123;
            console.log&#40;&#39;DOM ready&#39;&#41;;
        &#125;&#41;;
    &lt;&#47;script&gt;
&lt;&#47;body&gt;
&lt;&#47;html&gt;</pre>
</body>
</html>"
`;

exports[`Format Snapshot Tests > Markdown Format Snapshots > should generate consistent markdown format for JavaScript files > javascript-files.md 1`] = `
"# Generated Project Fusion File

**Project:** snapshot-test

**Generated:** [TIMESTAMP]

**Files:** 2

**Generated by:** [project-fusion v1.0.0](https://github.com/the99studio/project-fusion)

---

## 📁 Table of Contents

- [index.js](#indexjs)
- [utils.js](#utilsjs)

---

## 📄 index.js {#indexjs}

\`\`\`javascript
// Main application entry point
const express = require('express');
const app = express();

app.get('/', (req, res) => {
    res.send('Hello, World!');
});

app.listen(3000, () => {
    console.log('Server running on port 3000');
});
\`\`\`

## 📄 utils.js {#utilsjs}

\`\`\`javascript
// Utility functions
function formatDate(date) {
    return date.toISOString().split('T')[0];
}

function capitalize(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
}

module.exports = { formatDate, capitalize };
\`\`\`

"
`;

exports[`Format Snapshot Tests > Markdown Format Snapshots > should generate consistent markdown format for TypeScript files > typescript-files.md 1`] = `
"# Generated Project Fusion File

**Project:** snapshot-test

**Generated:** [TIMESTAMP]

**Files:** 2

**Generated by:** [project-fusion v1.0.0](https://github.com/the99studio/project-fusion)

---

## 📁 Table of Contents

- [service.ts](#servicets)
- [types.ts](#typests)

---

## 📄 service.ts {#servicets}

\`\`\`typescript
// User service
import { User, CreateUserRequest, UserRole } from './types.js';

export class UserService {
    private users: User[] = [];
    private nextId = 1;

    createUser(request: CreateUserRequest): User {
        const user: User = {
            id: this.nextId++,
            name: request.name,
            email: request.email,
            createdAt: new Date()
        };
        
        this.users.push(user);
        return user;
    }

    findUser(id: number): User | undefined {
        return this.users.find(user => user.id === id);
    }
}
\`\`\`

## 📄 types.ts {#typests}

\`\`\`typescript
// Type definitions
export interface User {
    id: number;
    name: string;
    email: string;
    createdAt: Date;
}

export type UserRole = 'admin' | 'user' | 'moderator';

export interface CreateUserRequest {
    name: string;
    email: string;
    role?: UserRole;
}
\`\`\`

"
`;

exports[`Format Snapshot Tests > Markdown Format Snapshots > should generate consistent markdown format for mixed file types > mixed-files.md 1`] = `
"# Generated Project Fusion File

**Project:** snapshot-test

**Generated:** [TIMESTAMP]

**Files:** 3

**Generated by:** [project-fusion v1.0.0](https://github.com/the99studio/project-fusion)

---

## 📁 Table of Contents

- [config.json](#configjson)
- [README.md](#readmemd)
- [script.sh](#scriptsh)

---

## 📄 config.json {#configjson}

\`\`\`json
{
  "name": "test-project",
  "version": "1.0.0",
  "dependencies": {
    "express": "^4.18.0",
    "typescript": "^5.0.0"
  },
  "scripts": {
    "start": "node index.js",
    "build": "tsc"
  }
}
\`\`\`

## 📄 README.md {#readmemd}

\`\`\`markdown
# Test Project

This is a test project for snapshot testing.

## Features

- **Fast**: Built with performance in mind
- **Secure**: Follows security best practices
- **Reliable**: Comprehensive test coverage

## Installation

\`\`\`bash
npm install
npm start
\`\`\`

## License

MIT License
\`\`\`

## 📄 script.sh {#scriptsh}

\`\`\`bash
#!/bin/bash
# Deployment script

echo "Starting deployment..."

# Build the project
npm run build

# Run tests
npm test

# Deploy to production
echo "Deploying to production..."
rsync -av dist/ user@server:/var/www/app/

echo "Deployment complete!"
\`\`\`

"
`;
